src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -242,6 +241,82 @@ public void RangeChangeNotification(IRingRange old, IRingRange now, bool increas
 
         #region Internal implementation methods
 
+        private async Task StartInBackground()
+        {
+            if (!await DoInitialLoadWithRetries()) return;
```

Comment:
Well, this goes with your other suggestion, that at some point we consider this non transient and we fail the calls from users by throwing (fast)  a concrete exception, not just a timeout after waiting for something that will most likely never complete. 
Of course I could also fail fast with the last exception but still retry forever.  That might not be that bad, and would keep logging this error frequently so that it doesn't go unnoticed.